File: clinic-backend/src/main/java/com/clinicapp/backend/dto/core/AppointmentDTO.java
package com.clinicapp.backend.dto.core;

import com.clinicapp.backend.model.core.AppointmentStatus;
import jakarta.validation.constraints.Future;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

/**
 * Data Transfer Object for Appointment information.
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class AppointmentDTO {

    private Long id;

    @NotNull(message = "Patient ID cannot be null")
    private Long patientId; // ID of the patient

    @NotNull(message = "Doctor ID cannot be null")
    private Long doctorId; // ID of the doctor (User ID)

    @NotNull(message = "Appointment start time cannot be null")
    @Future(message = "Appointment must be in the future")
    private LocalDateTime startTime;

    @NotNull(message = "Appointment end time cannot be null")
    @Future(message = "Appointment must be in the future")
    private LocalDateTime endTime;

    @NotBlank(message = "Room cannot be blank")
    private String room;

    @NotNull(message = "Status cannot be null")
    private AppointmentStatus status;

    private String notes;

    // Optional fields to include patient/doctor names in response for convenience
    private String patientFirstName;
    private String patientLastName;
    private String doctorFirstName;
    private String doctorLastName;

    // Timestamps (usually not needed in request DTOs, but useful in responses)
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}

File: clinic-frontend1/src/app/models/appointment.dto.ts
import { AppointmentStatus } from "./appointment-status.enum"; // Import the enum

// Matches com.clinicapp.backend.dto.core.AppointmentDTO
export interface AppointmentDTO {
  id?: number; // Optional for creation
  patientId: number;
  doctorId: number;
  startTime: string; // Use string for ISO date format
  endTime: string;   // Use string for ISO date format
  room?: string;
  status: AppointmentStatus; // Use the imported enum
  notes?: string;
  patientFirstName?: string; // Read-only, provided by backend
  patientLastName?: string;  // Read-only
  doctorFirstName?: string;  // Read-only
  doctorLastName?: string;   // Read-only
  createdAt?: string;        // Read-only
  updatedAt?: string;        // Read-only
}